driver/src/solution_submission/mod.rs

```diff
@@ -131,7 +131,9 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
 fn extract_transaction_hash(err: &Error) -> Option<H256> {
     err.downcast_ref::<MethodError>()
         .and_then(|method_error| match &method_error.inner {
-            ExecutionError::Failure(tx_hash) => Some(*tx_hash),
+            ExecutionError::Failure(transaction_receipt) => {
```

Comment:
Would this be the task related to the fact that ethcontracts now returns transactions receipts for failed transactions? If so, I believe it would be my responsibility to create a task here that makes use of this returned receipt.

Reply:
Yes, that would be great. We can quickly tackle this one outside of this PR.

Reply:
Once the next ethcontract version has been released we can move this out to a separate "Update ethonctract" PR. I wanted to include it here to make the build succeed.

Reply:
Already done! #691